phoenix-core/src/it/java/org/apache/phoenix/end2end/EncodeBinaryFunctionIT.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.end2end;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+import org.apache.phoenix.exception.SQLExceptionCode;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+
+@Category(ParallelStatsDisabledTest.class)
+public class EncodeBinaryFunctionIT extends ParallelStatsDisabledIT {
+
+    @Test
+    public void testEncodeHex() throws Exception {
+        Connection conn = DriverManager.getConnection(getUrl());
+
+        String testTable = generateUniqueName();
+        String ddl = "CREATE TABLE " + testTable + " (id INTEGER PRIMARY KEY, data VARBINARY)";
+
+        conn.createStatement().execute(ddl);
+
+        PreparedStatement ps = conn.prepareStatement(
+                "UPSERT INTO " + testTable + " (id, data) VALUES (1, X'48656c6c6f50686f656e6978')");
+        ps.execute();
+        conn.commit();
+
+        ResultSet rs = conn.createStatement().executeQuery("SELECT ENCODE_BINARY(data, 'HEX') FROM " + testTable);
+        assertTrue(rs.next());
+        String actualString = rs.getString(1);
+        assertEquals("48656c6c6f50686f656e6978", actualString);
```

Comment:
nit: use a java constant for these kinds of tests, with toLower() toUpper() if needed